backend/src/neuroagent/tools/literature_search_tool.py

```diff
@@ -84,22 +88,55 @@ async def arun(self) -> str:
         )
 
         # Prepare the request's body
-        req_body = {
-            "query": self.input_schema.query,
-            "retriever_k": self.metadata.retriever_k,
-            "use_reranker": self.metadata.use_reranker,
-            "reranker_k": self.metadata.reranker_k,
-        }
-
+        req_body = self.create_query(
+            query=self.input_schema.query,
+            article_types=self.input_schema.article_types,
+            authors=self.input_schema.authors,
+            journals=self.input_schema.journals,
+            date_from=self.input_schema.date_from,
+            date_to=self.input_schema.date_to,
+            retriever_k=self.metadata.retriever_k,
+            use_reranker=self.metadata.use_reranker,
+            reranker_k=self.metadata.reranker_k,
+        )
         # Send the request
         response = await self.metadata.httpx_client.get(
             self.metadata.literature_search_url,
             headers={"Authorization": f"Bearer {self.metadata.token}"},
-            params=req_body,  # type: ignore
+            params=req_body,
             timeout=None,
         )
+        if response.status_code != 200:
+            return response.json()
+        else:
+            return self._process_output(response.json())
+
+    @staticmethod
+    def create_query(
+        query: str,
+        article_types: list[str] | None,
+        authors: list[str] | None,
+        journals: list[str] | None,
+        date_from: str | None,
+        date_to: str | None,
+        retriever_k: int | None,
+        reranker_k: int | None,
+        use_reranker: bool | None,
+    ) -> dict[str, str | int | list[str]]:
```

Comment:
So why don't we use the autogenerate types even more and have this create_query function return an instance of
`RetrievalRequest(GenerativeQARequest)`?

That way we will have static type checking both for the request and response bodies

Reply:
Wait wait..the above does not make any sense..the `/retrieval` is a GET..there is no request body...everything is a query parameter...but maybe the `RetrievalRequest` model mirrors the query parameters ...please check..in that case